Update the README with Panama specific information, and information about the repo (#51)
openjdk/panama-foreign: README.md

> 

Reviewer: In general, I'd prefer to replace `repo` with `repository`; I'd also prefer to use a less informal approach (e.g. instead of "you can build the repo", "the repo can be built")

Welcome to the JDK!
===================

For information about building the JDK, including how to retrieve all
of the source code, please see either of these files:

  * doc/building.html   (html version)
  * doc/building.md     (markdown version)

See http://openjdk.java.net/ for more information about the OpenJDK
Community and the JDK.

---
About Project Panama
===================

> This is a repo of OpenJDK Project Panama: Interconnecting JVM and native code. You can find more information at: https://openjdk.java.net/projects/panama/

Reviewer: This repository contains changes which aims at improving the interoperability between the Java programming language and native libraries, which one of the main goals of Project Panama. More information can be found at: https://openjdk.java.net/projects/panama/

> This repository contains changes which aims at improving the interoperability between the Java programming language and native libraries, which one of the main goals of Project Panama. More information can be found at: https://openjdk.java.net/projects/panama/

Author: ```suggestion
This repository contains changes which aim at improving the interoperability between the Java programming language and native libraries, which is one of the main goals of Project Panama. More information can be found at: https://openjdk.java.net/projects/panama/
```

> You can find the Project Panama JBS dashboard at: https://bugs.openjdk.java.net/Dashboard.jspa?selectPageId=18412

Reviewer: The Project Panama JBS dashboard can be found at

> You can find early access (EA) snapshots at: http://jdk.java.net/panama/

Reviewer: Early acccess (EA) binary snapshot can be found at:

> Early acccess (EA) binary snapshot can be found at: http://jdk.java.net/panama/

Author: ```suggestion
Early acccess (EA) binary snapshots can be found at: http://jdk.java.net/panama/
```

About the panama-foreign repository
===================
This repository houses three main branches:

> - **foreign-memaccess**: Contains the developement of the foreign memory access API for [JEP 370](https://openjdk.java.net/jeps/370), which can be used to interact with different kinds of memory resource, inluding so-called off-heap or native memory.

Author: ```suggestion
- **foreign-memaccess**: Contains the developement of the foreign memory access API for [JEP 370](https://openjdk.java.net/jeps/370), which can be used to interact with different kinds of memory resources, including so-called off-heap or native memory.
```

> - **foreign-abi**: Contains the developement of the foreign function interface (FFI), which can be used to call native code in a  .dll/.so/.dylib, or to create a native function pointer to a Java method which can be passed to code in a native library.

Author: ```suggestion
- **foreign-abi**: Contains the developement of the foreign function interface (FFI), which can be used to call native code in a .dll/.so/.dylib, or to create a native function pointer to a Java method which can be passed to code in a native library.
```

> - **foreign-jextract**: Contains the developement of the jextract API, which can be used to create an abstract syntax tree from a C header file. There's also the accompanying jextract tool, which is a wrapper for the API and can be used to generate a Java interface to call the C library described by a given header file.

Reviewer: Contains the development of an API to parse native headers, which can be used to create an abstract representation (declarations) from a C header file. This branch also provides an accompanying extraction tool (jextract), which is built on top of the API, and can be used to generate *Java bindings* to access functions and/or structs in a native library described by a given header file.

> The foreign-jextract branch depends on the foreign-abi branch, which in turn depends on the foreign-memaccess branch. As such, the foreign-jextract is the most complete of the three and therefore the default branch of this repositorysitory.

Author: ```suggestion
The foreign-jextract branch depends on the foreign-abi branch, which in turn depends on the foreign-memaccess branch. As such, the foreign-jextract branch is the most complete of the three and therefore the default branch of this repository.
```

Building notes
===================

> Jextract requires [LLVM 9.x](https://releases.llvm.org/download.html) as a dependency. So, if you want to build the foreign-jextract branch yourself you will need to pass `--with-libclang=/path/to/llvm/root` when generating a build configuration. Please see [doc/building.md](https://github.com/openjdk/panama-foreign/blob/foreign-jextract/doc/building.md) for general building instructions.

Reviewer: So, to build the foreign-jextract branch, the flag `--with-libclang=/path/to/llvm/root` must be supplied when generating a build configuration.

> Jextract requires [LLVM 9.x](https://releases.llvm.org/download.html) as a dependency. So, to build the foreign-jextract branch, the flag --with-libclang=/path/to/llvm/root must be supplied when generating a build configuration. Please see [doc/building.md](https://github.com/openjdk/panama-foreign/blob/foreign-jextract/doc/building.md) for general building instructions.

Author: ```suggestion
Jextract requires [LLVM 9.x](https://releases.llvm.org/download.html) as a dependency. So, to build the foreign-jextract branch, the flag `--with-libclang=/path/to/llvm/root` must be supplied when generating a build configuration. Please see [doc/building.md](https://github.com/openjdk/panama-foreign/blob/foreign-jextract/doc/building.md) for general building instructions.
```
